Implement simplified surfaces and faster contouring

bclyons12 opened the pull request:
This creates a simplified, lighter-weight version of flux surfaces suited to some applications (e.g., FRESCO). It also uses the new IMASutils contouring where applicable.

bclyons12 commented:
@orso82 I didn't modify `flux_surface()` since that has lots of fancy logic for open surfaces that I can't guarantee would be covered by IMASutil's contouring. It looks like we could do the same thing for `trace_surfaces()` that I did for `trace_simple_surfaces()`, and for the other version of `find_psi_boundary()` like I did for my version.

bclyons12 commented:
Also, once this PR is deemed ready to merge, this is a circumstance where we should run all the FUSE tests on it first. That will stress test the new contouring routines.

orso82 commented:
To run the FUSE tests using your changes, just open a PR in FUSE on a branch with the same `simple_surface` name that you have used here.

bclyons12 commented:
The IMAS 2.0 refactoring has been merged into this, which means both versions of `find_psi_boundary()` now use the new contouring.

bclyons12 commented:
@orso82 `trace_surfaces()` now uses the new IMASutils contouring as well. Only `flux_surface` (and other fucntions that depend on that) is left untouched because of the other types of surfaces that it can contour.

bclyons12 commented:
@orso82 All the FUSE regressions pass (https://github.com/ProjectTorreyPines/FUSE.jl/pull/770) so I'll merge this unless you want to make the `find_psi_boundary` more robust under this PR.

orso82 commented:
@bclyons12 this is great, although it's unclear to me what's the difference between `trace_surfaces` and `trace_simple_surfaces`.

Do we really need both?

If we can't just have a single version, could you add a description to the docstring of`trace_simple_surfaces` and perhaps highlight the difference with respect to `trace_surfaces`

bclyons12 commented:
@orso82 I cleaned up `trace_simple_surfaces` even more and added some documentation. The idea is that the FluxSurface struct has a lot of information, but if all we want is to perform flux surface averaging, we can get away with a lot less. In FRESCO where I might need to trace surfaces every iteration, this reduces the computation needed.

orso82 commented on src/physics/fluxsurfaces.jl:
```diff
@@ -1613,18 +1613,24 @@ function flux_surfaces(eqt::equilibrium__time_slice{T1}, wall_r::AbstractVector{
-    ig, jg = pts[argmax(pts_in_wall)] # this are the indices for the guess inside the wall
-    rguess, zguess = r[ig], z[jg]
-    RA, ZA = find_magnetic_axis(r, z, PSI_interpolant, psi_sign; rguess, zguess)
+    if !isempty(wall_r)
```
Thank you for doing this @bclyons12 ! 👍 

I wonder if we should add this logic to the `find_magnetic_axis()` function.

bclyons12 replied:
You would have to pass the wall to it then, so up to you. You could make a separate function that takes the wall as an argument and does this I suppose.

bclyons12 commented:
@orso82 If the FUSE tests pass (https://github.com/ProjectTorreyPines/FUSE.jl/pull/770), I would like to merge this.

bclyons12 commented:
All FUSE tests pass (https://github.com/ProjectTorreyPines/FUSE.jl/pull/770)
